modules/imgproc/src/warp_kernels.simd.hpp

```diff
@@ -3226,31 +3300,46 @@ void warpAffineLinearInvoker_8UC3(const uint8_t *src_data, size_t src_step, int
             for (; x <= dstcols - uf; x += uf) {
                 // [TODO] apply halide trick
                 CV_WARPAFFINE_VECTOR_COMPUTE_MAPPED_COORD2(LINEAR, C3);
-    #if defined(CV_NEON_AARCH64) && CV_NEON_AARCH64
-                uint8x8_t p00r, p01r, p10r, p11r,
-                          p00g, p01g, p10g, p11g,
-                          p00b, p01b, p10b, p11b;
-    #endif
                 if (v_reduce_min(inner_mask) != 0) {
-    #if defined(CV_NEON_AARCH64) && CV_NEON_AARCH64
-                    CV_WARP_VECTOR_LINEAR_SHUFFLE_ALLWITHIN_NEON_U8(C3);
-    #else
-                    CV_WARP_VECTOR_SHUFFLE_ALLWITHIN(LINEAR, C3, 8U);
+                    float valpha[max_uf], vbeta[max_uf];
+                    vx_store(valpha, src_x0);
+                    vx_store(valpha+vlanes_32, src_x1);
+                    vx_store(vbeta, src_y0);
+                    vx_store(vbeta+vlanes_32, src_y1);
+    #if CV_SIMD256
+                    CV_WARP_VECTOR_SHUFFLE_INTER_STORE(SIMD256, LINEAR, 8U, C3);
+    #elif CV_SIMD128
+                    CV_WARP_VECTOR_SHUFFLE_INTER_STORE(SIMD128, LINEAR, 8U, C3);
+    #elif CV_SIMD_SCALABLE
+                    CV_WARP_VECTOR_SHUFFLE_INTER_STORE(SIMDX, LINEAR, 8U, C3);
+                    // int32_t tmp_buf[max_vlanes_16 * 4];
+                    // for (int k = 0; k < uf; k++) {
+                    //     const uint8_t *srcptrk = src + addr[k];
+                    //     v_float32 ik_fpix0 = v_cvt_f32(v_reinterpret_as_s32(v_load_expand_q<4>(srcptrk)));
+                    //     v_float32 ik_fpix1 = v_cvt_f32(v_reinterpret_as_s32(v_load_expand_q<4>(srcptrk + 3)));
+                    //     v_float32 ik_fpix2 = v_cvt_f32(v_reinterpret_as_s32(v_load_expand_q<4>(srcptrk + srcstep)));
+                    //     v_float32 ik_fpix3 = v_cvt_f32(v_reinterpret_as_s32(v_load_expand_q<4>(srcptrk + srcstep + 3)));
+                    //     v_float32 ik_alpha = vx_setall_f32(valpha[k]);
+                    //     v_float32 ik_beta = vx_setall_f32(vbeta[k]);
+
+                    //     ik_fpix0 = v_fma(ik_alpha, v_sub(ik_fpix1, ik_fpix0), ik_fpix0);
+                    //     ik_fpix2 = v_fma(ik_alpha, v_sub(ik_fpix3, ik_fpix2), ik_fpix2);
+                    //     ik_fpix0 = v_fma(ik_beta,  v_sub(ik_fpix2, ik_fpix0), ik_fpix0);
+                    //     auto ik_pix0 = v_round(ik_fpix0);
+                    //     v_store<4>(tmp_buf + 3 * k, ik_pix0);
+                    // }
+                    // for (int k = 0; k < uf * 3; k++) {
+                    //     dstptr[3 * x + k] = saturate_cast<uchar>(tmp_buf[k]);
+                    // };
```

Comment:
Please remove the dead code, if it's not needed.

Reply:
Done.